instrumentation/jdbc/src/main/java/com/solarwinds/opentelemetry/instrumentation/JdbcConnectionInstrumentation.java

```diff
@@ -69,10 +69,13 @@ public ElementMatcher<TypeDescription> typeMatcher() {
   @Override
   public void transform(TypeTransformer transformer) {
     transformer.applyAdviceToMethod(
-        nameStartsWith("prepare")
-            .and(takesArgument(0, String.class))
-            // Also include CallableStatement, which is a subtype of PreparedStatement
-            .and(returns(implementsInterface(named("java.sql.PreparedStatement")))),
+        named("prepareCall")
+            .and(takesArgument(0, String.class).and(takesArgument(1, int.class)))
+            .and(returns(implementsInterface(named("java.sql.PreparedStatement"))))
```

Comment:
curious why we still need the condition in lines 72-24. my understanding is that `java.sql.prepareCall` would return a `CallableStatement` which is a subinterface of `PreparedStatement`... which would not match despite original comment(?) and thus not apply the advice. guess i'm wondering why we still need to specify `prepareCall` at all.

Reply:
> curious why we still need the condition in lines 72-24. my understanding is that java.sql.prepareCall would return a CallableStatement which is a subinterface of PreparedStatement... which would not match despite original comment(?) and thus not apply the advice.

You're correct. `CallableStatement` is a subinterface of `PreparedStatement`, therefore anything that implements `CallableStatement` also implements  `PreparedStatement`. So it should match the test. 

> guess i'm wondering why we still need to specify prepareCall at all.

It's specified because there can be `CallableStatement` that aren't calling stored procedure and those take additional arguments with `int` being the second arg.

Reply:
hmm, from https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html, `prepareCall` is specifically for calling stored procedures. i think it may be an edge case that a client would somewhat abuse it to make prepared statements, or are you seeing this in the older hibernate framework?

and now i'm wondering, in the case where `prepareCall` is used for a `CallableStatement` meant to call an actual stored procedure--how are we managing to skip instrumenting it?

finally, is the `takesArgument` condition matching for the argument to `prepareCall` and `prepareStatement`? if so... now that i'm looking at the JDBC reference, there are quite a few other prepareStatement signatures (>.<), should we try to cover them too?

Reply:
> i think it may be an edge case that a client would somewhat abuse it to make prepared statements, or are you seeing this in the older hibernate framework?

Possibly, however I assumed only the first signature only applies to stored procedure call due it's description being the only one that mentions procedure call.

> and now i'm wondering, in the case where prepareCall is used for a CallableStatement meant to call an actual stored procedure--how are we managing to skip instrumenting it?

`CallableStatement` description specifically states it's for stored procedure calls so any of those methods can be safely assumed to be calling a stored procedure. I'll remove it.

> finally, is the takesArgument condition matching for the argument to prepareCall and prepareStatement?

Just `prepareCall` and `prepareStatement` is matched by name.